apps/accounts/tests/test_admin.py

```diff
@@ -149,6 +146,22 @@ def test_get_inlines_non_staff(
         assert ac_admin.DatacenterNoteInline not in inlines
         assert "Associated hosting providers" not in fieldset_names
 
+    def test_list_of_users_who_can_manage_dc_is_displayed(
+        self, db, sample_hoster_user, greenweb_staff_user, datacenter
+    ):
+        # given: 2 distinct users have explicit permissions to manage the datacenter
+        assign_perm(str(manage_datacenter), sample_hoster_user, datacenter)
+        assign_perm(str(manage_datacenter), greenweb_staff_user, datacenter)
+
+        # when the datacenter admin page is accessed by one of the users
```

Comment:
This is looks like a really nifty way to test Django Admin functionality without needing to faff with creating a client, logging them in and so on (which is the way I would have thought I had to test it)

Is there a blog post or page you might point to explain this in more detail?

It's ok if not, but I this looks like the first instance iv'e seen of instantiating the admin site like this.

Reply:
I literally copied it from the test above... which you added/modified :grin: 
https://github.com/thegreenwebfoundation/admin-portal/pull/481/files#diff-030e218c1ff710051a2f710b11bc6290c4d32c8f15e6f4c1b6cdcce9028c1877R125-R127
But I agree it's very handy!